leshan-core/src/test/java/org/eclipse/leshan/core/request/argument/ArgumentsParsingTest.java

```diff
@@ -112,4 +112,10 @@ public void should_not_parse_text_into_argument_with_value_without_equal() throw
     public void should_not_parse_text_into_argument_with_value_without_quotes() throws InvalidArgumentException {
         Arguments.parse("3=hello");
     }
+
+    @Test(expected = InvalidArgumentException.class)
+    public void should_not_parse_unquoted_value() throws InvalidArgumentException {
+        Arguments.parse("3=string,4='Value'");
+    }
+
```

Comment:
That's great you find this issue. I make test for different behaviours separately to check if implementation is valid but I don't check this kind behaviours together :clap:

Reply:
I suspected it reading the code :slightly_smiling_face: